Kitodo/src/main/java/org/kitodo/production/services/dataeditor/DataEditorService.java

```diff
@@ -227,25 +231,40 @@ public static StructuralElementViewInterface getStructuralElementView(DataEditor
         Optional<LogicalDivision> selectedStructure = dataEditor.getSelectedStructure();
         if (selectedStructure.isPresent()) {
             return dataEditor.getRulesetManagement()
-                    .getStructuralElementView(
-                            selectedStructure.get().getType(),
+                    .getStructuralElementView(selectedStructure.get().getType(),
                             dataEditor.getAcquisitionStage(), dataEditor.getPriorityList());
-        } else {
-            TreeNode selectedLogicalNode = dataEditor.getStructurePanel().getSelectedLogicalNode();
-            if (Objects.nonNull(selectedLogicalNode)
-                    && selectedLogicalNode.getData() instanceof StructureTreeNode) {
-                StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
-                if (structureTreeNode.getDataObject() instanceof View) {
-                    View view = (View) structureTreeNode.getDataObject();
-                    if (Objects.nonNull(view.getPhysicalDivision())) {
-                        return dataEditor.getRulesetManagement().getStructuralElementView(
-                            view.getPhysicalDivision().getType(),
-                                dataEditor.getAcquisitionStage(), dataEditor.getPriorityList());
-                    }
-                }
+        }
+
+        TreeNode selectedLogicalNode = dataEditor.getStructurePanel().getSelectedLogicalNode();
+        if (Objects.isNull(selectedLogicalNode)) {
+            throw new IllegalStateException("No logical node selected");
+        }
+
+        if (!(selectedLogicalNode.getData() instanceof StructureTreeNode)) {
+            throw new IllegalStateException("Selected logical node data is of bad type: "
+                    + (Objects.isNull(selectedLogicalNode.getData()) ? "null"
+                            : selectedLogicalNode.getData().getClass().getName()));
+        }
+        StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
+
+        if (structureTreeNode.getDataObject() instanceof View) {
+            View view = (View) structureTreeNode.getDataObject();
+            if (Objects.isNull(view.getPhysicalDivision())) {
+                throw new IllegalStateException("View has no physical division assigned");
             }
+
+            return dataEditor.getRulesetManagement().getStructuralElementView(
+                    view.getPhysicalDivision().getType(),
+                        dataEditor.getAcquisitionStage(), dataEditor.getPriorityList());
         }
-        throw new IllegalStateException();
+
+        if (structureTreeNode.getDataObject() instanceof Process) {
+            return null;
+        }
+
+        throw new IllegalStateException("Data object is of bad type: "
```

Comment:
It seems to me this exception is always thrown! I get this exception when selecting the _parent_ process of the current process in the metadata editor (same as before, no problem with linked child processes):

https://user-images.githubusercontent.com/19183925/149781042-9ad6e51f-481e-41ea-9b75-33e31fc37abe.mov

Reply:
There is literally nothing happening in the video.

The exception happened because the `dataObject` is of type `Process` for parent processes, and the _master_ expects it to be a `View`. But this is now checked for (line 261). So I don’t understand why the exception still happens for you.

The exception should print the "bad" object type now. What does it print for you?

Reply:
@solth Thank you for catching! Looks like I didn’t get the error yesterday, probably because the process directory of my parent process is missing on disk… 🙄